core/node/eth_sender/src/aggregator.rs

```diff
@@ -178,6 +188,38 @@ impl Aggregator {
         }
     }
 
+    async fn get_or_init_tree(&mut self) -> &mut MiniMerkleTree<L1Tx> {
+        if self.priority_merkle_tree.get().is_none() {
+            // We unwrap here since it is only invoked during initialization
+            let mut connection = self.pool.connection_tagged("eth_sender").await.unwrap();
+
+            let priority_op_hashes =
+                if let Some(priority_tree_start_index) = self.config.priority_tree_start_index {
+                    // We unwrap here since it is only invoked during initialization
+                    connection
+                        .transactions_dal()
+                        .get_l1_transactions_hashes(priority_tree_start_index)
+                        .await
+                        .unwrap()
+                } else {
+                    vec![]
+                };
+            let priority_merkle_tree = MiniMerkleTree::<L1Tx>::from_hashes(
+                KeccakHasher,
+                priority_op_hashes.into_iter(),
+                None,
+            );
+
+            // The error can only be thrown if this tree has been populated before.
+            // It is expected that this tree is populated deterministically based on config and
+            // so if some other thread populated it, the stored value should've been the same.
+            let _ = self.priority_merkle_tree.set(priority_merkle_tree);
```

Comment:
IIUC, other threads cannot have populated the tree because of `&mut self` receiver; if multiple threads could acquire such a receiver in parallel, that'd violate fundamental Rust invariants. (For the same reason, it looks like `Option<_>` would be sufficient for the tree, as I've noted in a comment above.)